pyulog/core.py

```diff
@@ -537,6 +583,10 @@ def _read_file_data(self, message_name_filter_list, read_until=None):
             while True:
                 data = self._file_handle.read(3)
                 header.initialize(data)
+                # try recovery with sync sequence in case of unknown msg_type
+                if ((header.msg_type_str == "UNKNOWN") and self._has_sync):
+                    if self._find_sync():
```

Comment:
Comments picked from https://github.com/PX4/pyulog/pull/41

> @bkueng: This changes existing behavior: newly added messages will be treated as corruption by older parsers. The current parser would just ignore those. I'm aware that we cannot have both perfectly w/o CRC. There's already existing logic to handle this.
Please use and integrate with the logic from here to check for corruption: https://github.com/PX4/pyulog/blob/master/pyulog/core.py#L603.

Reply:
I does not change the behavior when there is no sync message (which is true for existing code out there), newly added messages will go through this block ones, not find the sync bytes, and fallback to existing error handling code.

There is a change in behavior for tackling new message_type messages when there exists a sync byte sequence in the file, which chooses to skip to the sync sequence rather than jump to a random number of bytes assuming that there is no corruption (the 10K size check you linked there is rather hacky and would snowball). I did this because that is the only sure way of handling this situation, which is precisely what the sync sequence is there for. The unknown sync byte error is printed out, so the user can update/add handling for it.

Reply:
Comments picked from https://github.com/PX4/pyulog/pull/41

> @bkueng: It's not random if it's not corrupt. And while it's an approximation, it has proven to work in practice.
Your change breaks backwards compatibility: logs with newly added message types will be regarded as corrupt. This will make additions like your tagged logging message more difficult in future. ULog has ways of dealing with it (there are incompatibility flags to refuse parsing, but it will make the update process more difficult and will force all users to update). I'm not saying we cannot do that, but I like the current behavior, and I currently don't see that we have to give that up.

Reply:
This would mean that the logic would keep ignoring sync bytes until one of the msg_size pair of bytes would resolves to something greater than 10k.

Reply:
If we also looked for sync bytes while 'ignoring'/skipping over msg_size bytes of an unknown msg_type packet, it'd solve both our problems. If the 'contents' of the unknown msg_type packet do not contain the sync bytes, simply ignore the packet and do not declare it corrupt, if they do, stop skipping over the bytes at the end of sync byte seq., declare corrupt, and restart parsing again from the next byte.

Reply:
Yes that's a workable approach.

Reply:
> Yes that's a workable approach.

1d913490118d59b396f82855bbbfb156dd116c63 does it.